### PR TITLE
VIH-9464 - handle of the user api exception 409 and retry

### PR DESCRIPTION
**Pull request by marcogagliardi**
https://tools.hmcts.net/jira/browse/VIH-9464

handle of the user api exception 409 and retry

### CODE REVIEW

File: BookingQueueSubscriber/BookingQueueSubscriber.Services/UserApi/UserService.cs [hunk 1]
```diff
@@ -37,20 +38,48 @@ public UserService(IUserApiClient userApiClient, ILogger<UserService> logger, IF
...
+                        {
+                            _logger.LogInformation(
+                                "User with contact email {contactEmail} does not exist. Creating an account. Second try.",
+                                contactEmail);
+                            return null;
+                        }
+                    }
+                    else
+                    {
+                        return null;
```
Comment (oliver-scott) [inline BookingQueueSubscriber/BookingQueueSubscriber.Services/UserApi/UserService.cs:80]: I think we said we were going to use `throw` here in order to preserve the original exception. Returning null here will cause a null reference exception for the caller when it tries to set the username, which would mask the original error.

`participant.Username = user.UserName;`

File: BookingQueueSubscriber/BookingQueueSubscriber.Services/UserApi/UserService.cs [hunk 2]
```diff
@@ -37,20 +38,48 @@ public UserService(IUserApiClient userApiClient, ILogger<UserService> logger, IF
...
+                    if (e.StatusCode == (int) HttpStatusCode.Conflict)
+                    {
+                        Thread.Sleep(1000);
+                        userProfile = await GetUserByContactEmail(contactEmail);
+                        if (userProfile == null)
+                        {
+                            _logger.LogInformation(
+                                "User with contact email {contactEmail} does not exist. Creating an account. Second try.",
+                                contactEmail);
+                            return null;
```
Comment (oliver-scott) [inline BookingQueueSubscriber/BookingQueueSubscriber.Services/UserApi/UserService.cs:75]: As per previous comment

File: BookingQueueSubscriber/BookingQueueSubscriber.Services/UserApi/UserService.cs (hunk 1 above)
Comment (marcogagliardi) [reply BookingQueueSubscriber/BookingQueueSubscriber.Services/UserApi/UserService.cs:80]: I changed the parent and it checks for null. 
On UserCreationAndNotification.cs in CreateUserAndSendNotificationAsync it is already checking for user null from the beginning. I thought that it is better to be consistent in the method. What I am saying is that it is already returning a null user in the original code.

Comment (oliver-scott) [reply BookingQueueSubscriber/BookingQueueSubscriber.Services/UserApi/UserService.cs:80]: oops sorry didn't see that you'd changed the parent, my bad

**Review by oliver-scott — APPROVED** [review]

**Pull request merged**